src/convert.cpp

```diff
@@ -61,48 +73,58 @@ ddwaf_object* to_ddwaf_object_object(
   Napi::Object obj,
   int depth,
   bool lim,
-  bool ignoreToJSON
+  bool ignoreToJSON,
+  JsSet jsSet
 ) {
   if (!ignoreToJSON) {
     Napi::Value toJSON = obj.Get("toJSON");
     if (toJSON.IsFunction()) {
-      return to_ddwaf_object(object, env, toJSON.As<Napi::Function>().Call(obj, {}), depth, lim, true);
+      return to_ddwaf_object(object, env, toJSON.As<Napi::Function>().Call(obj, {}), depth, lim, true, jsSet);
     }
   }
 
-  Napi::Array properties = obj.GetPropertyNames();
-  uint32_t len = properties.Length();
-  if (lim && len > DDWAF_MAX_CONTAINER_SIZE) {
-    len = DDWAF_MAX_CONTAINER_SIZE;
-  }
-  if (env.IsExceptionPending()) {
-    mlog("Exception pending");
-    return nullptr;
-  }
-
-  ddwaf_object* map = ddwaf_object_map(object);
-  if (map == nullptr) {
-    mlog("failed to create map");
-    return nullptr;
-  }
+  if (jsSet.Has(obj)) {
+    mlog("Circular dependency")
+    return ddwaf_object_invalid(object);
+  } else {
```

Comment:
could we use the early exit return syntax instead of if/else, it makes the diff difficult to read :/

Reply:
Yes, and now it is a bit easier to compare the PR...